src/Lean/Data/Lsp/Extra.lean

```diff
@@ -35,8 +35,23 @@ instance : FromJson WaitForDiagnostics :=
 instance : ToJson WaitForDiagnostics :=
   ⟨fun o => mkObj []⟩
 
+inductive LeanFileProgressKind
+  | processing | fatalError
+  deriving Inhabited, BEq
```

Comment:
Is there a reason you opted for 1/2 instead of "processing"/"fatalError" as the json serialization?  There shouldn't be any performance issues with strings here, we already include the much longer file name and various keys in the message.

Reply:
Fwiw, LSP tends to encode enums as numbers so I guess it matches?

Reply:
Yeah, that was my reasoning too, I took `DiagnosticSeverity` as an example.

Reply:
Ok, this makes sense.